backend/ng/team/controllers/list_teams_in_event.py

```
@@ -1,49 +1,24 @@
 """
-/backend/ng/team/controllers/list_teams_in_event.py
 Retrieves a list of all teams within a specific event.
 """
 
 from typing import Any
 
-from ...event.models.Event import Event
+from ...core import NotFoundError
 from ..models.Team import Team
 
 
 def list_teams_in_event(event_id: int) -> dict[str, Any]:
-    """Gets all teams in a event with their basic info.
-
-    Args:
-        event_id (int): The event ID to list teams from.
+    """
+    Gets all teams in an event.
 
     Returns:
-        dict: Success status, list of teams with stats, and event info.
+        dict: Teams data for the event.
+
     """
-    event = Event.find_by_id(event_id)
-    if not event:
-        return {
-            "success": False,
-            "error": f"Event with ID {event_id} does not exist",
-        }
-
-    teams = Team.find_all_by_event(event_id)
-
-    team_list = []
-    for team in teams:
-        team_list.append(
-            {
-                "id": team.id,
-                "name": team.name,
-                "member_count": team.member_count,
-                "max_team_size": event.max_team_size,
-                "is_full": team.member_count >= event.max_team_size,
-                "invite_code": team.invite_code,
-                "ranked": team.ranked,
-            }
-        )
-
-    return {
-        "success": True,
-        "teams": team_list,
-        "event_name": event.name,
-        "total_teams": len(team_list),
-    }
+    teams_data = Team.get_all_teams_in_event_for_public(event_id)
+
+    if teams_data is None:
+        raise NotFoundError(f"Event with ID {event_id} does not exist")
```

Is this the correct error? Couldn't this mean that there is an event that just doesn't have any teams yet? And if it doesn't have any teams, why raise an error instead of just returning an empty array?